app/models/person.rb

```diff
@@ -288,6 +288,10 @@ def first_name_with_initial
   # Deprecated: This is view logic (how to display name) and thus should not be in model layer
   # Consider using PersonViewUtils
   def name(community_or_display_type=nil)
+    # TODO Make `community_or_display_type` param mandatory.
+    # The custom error is here in order to get better error message while, but it can be removed in the future.
+    raise ArgumentError.new("Missing community or display type. Can not know how to format the name") if community_or_display_type.nil?
+
```

Comment:
I can see your point but it leaves us with method signature that lies to user. From "code that documents itself" point of view method signature is very important. I would guess I look at that before looking at any comments or the method body, and probably often times only look at the signature and only study the body if I don't feel like I can guess what the method does from name, context and parameters.